lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -439,6 +457,12 @@ class ClassMetadataInfo implements ClassMetadata
      * - <b>nullable</b> (boolean, optional)
      * Whether the column is nullable. Defaults to FALSE.
      *
+     * - <b>'notInsertable'</b> (boolean, optional)
+     * Whether the column is not insertable. Optional. Is only set if value is TRUE.
+     *
+     * - <b>'notUpdatable'</b> (boolean, optional)
+     * Whether the column is updatable. Optional. Is only set if value is TRUE.
+     *
```

Comment:
I'd recommend against negative logic and do something like this:

```suggestion
     * - <b>'insertable'</b> (boolean, optional)
     * Whether the column is not insertable. Optional. Is only set if value is FALSE.
     *
     * - <b>'updatable'</b> (boolean, optional)
     * Whether the column is updatable. Optional. Is only set if value is FALSE.
     *
```

Reply:
Then the values have to be stored in the cached metadata for all fields, making the serialized cache string sizes larger and deserialize performance slower. The way it is now also simplifies checks for just having to test for isset() of the key.